### PR TITLE
Challenge/hosni/toh res

### PR DESCRIPTION
**Pull request by HosniMaRu**

### CODE REVIEW

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/components/HeroList/index.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import './heroeList.css';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import { loadHeroes, deleteHero, addHero } from '../../redux/actions/actionCreators';
+
+function List({ heroes, dispatch }) {
+  function handleAdd() {
+    const inputNewHero = document.getElementById('heroInputCreation');
+    const heroNew = { id: (heroes[heroes.length - 1].id + 1), name: inputNewHero.value };
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/components/HeroList/index.jsx:11]: Break into multiple lines to improve readability

Comment (gilbecao) [reply hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/components/HeroList/index.jsx:11]: `
function name({ heroes }) {
  return {
    heroes
  };
}
`

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  return async (dispatch) => {
+    try {
+      const { data } = await axios(url);
+      dispatch({
+        type: actionTypes.LOAD_HEROES,
+        heroes: data
+      });
+    } catch (error) {
+      dispatch({
+        type: 'LOAD_HEROES_ERROR'
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js:17]: Magic string

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+export function deleteHero(heroId) {
+  return async (dispatch) => {
+    await axios.delete(`${url}/${heroId}`);
+    dispatch({
+      type: actionTypes.DELETE_HERO,
+      heroId
+    });
+  };
+  // return {
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js:41]: Do not push commented code

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+export function updateHero(hero) {
+  return async (dispatch) => {
+    const { data } = await axios.put(`${url}/${hero.id}`, hero);
+    dispatch({
+      type: actionTypes.UPDATE_HERO,
+      hero: data
+    });
+  };
+  // return {
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js:55]: Remove commented code

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+export function loadHero(hero) {
+  return async (dispatch) => {
+    const { data } = await axios(`${url}/${hero.id}`);
+    dispatch({
+      type: actionTypes.LOAD_HERO,
+      hero: data
+    });
+  };
+  // return {
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js:69]: Remove commented code

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+export function getHeroById(heroId) {
+  const hero = HEROES.find((current) => current.id === +heroId);
+
+  return {
+    type: actionTypes.LOAD_HERO,
+    hero
+  };
+  // const hero = HEROES.find((current) => current.id === +heroId);
```
Comment (gilbecao) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.js:82]: Remove commented code

**Review by Laura-SM — APPROVED** [review]
Good job!

**Review by JDSanchon — APPROVED** [review]

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.test.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    expect(dispatch).toHaveBeenCalledWith({
+      type: actionTypes.LOAD_HEROES
+    });
+  });
+});
+describe('loadHeroesAxios', () => {
+  test('should dispatch LOAD_HEROES', async () => {
+    axios.mockResolvedValue({ data: ['Skylab mola moltisim'] });
+    const dispatch = jest.fn();
+    // act
```
Comment (ulldecorb) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.test.js:38]: Remember to delete forgotten comments :)

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.test.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    expect(dispatch).toHaveBeenCalledWith({
+      type: actionTypes.LOAD_HEROES,
+      heroes: ['Skylab mola moltisim']
+    });
+  });
+
+  test('should dispatch LOAD_HEROES_ERROR', async () => {
+    axios.mockRejectedValue();
+    const dispatch = jest.fn();
+    // act
```
Comment (ulldecorb) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/redux/actions/actionCreators.test.js:50]: Delete comment again

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/utils/test-utils.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/jsx-filename-extension */
+// test-utils.js
```
Comment (ulldecorb) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/utils/test-utils.js:3]: Delete comments on lines 3, 7, 30 & 32

File: hosni-marco/challenge/toh-fullstack/server/toh-rest-api/src/controllers/heroesController.test.js [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+  updateById,
+  deleteById
+} = require('./heroesController')();
+const Hero = require('../model/heroModel');
+
+jest.mock('../model/heroModel');
+
+describe('getAll', () => {
+  test('shoud get all heroes', async () => {
+    // arrange
```
Comment (ulldecorb) [inline hosni-marco/challenge/toh-fullstack/server/toh-rest-api/src/controllers/heroesController.test.js:14]: Delete comments all over the file...

File: hosni-marco/lecture/rest/toh-rest-api/src/controllers/heroesController.test.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
+const heroesController = require('./heroesController');
+
+describe('getAll', () => {
+  test('shoud get all heroes', () => {
+    // arrange
```
Comment (ulldecorb) [inline hosni-marco/lecture/rest/toh-rest-api/src/controllers/heroesController.test.js:5]: Delete comments....

**Review by JDSanchon — APPROVED** [review]

File: hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/App.test.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+// import { render, screen } from '@testing-library/react';
+// import App from './App';
+
+// test('renders learn react link', () => {
+//   render(<App />);
+//   const linkElement = screen.getByText(/learn react/i);
+//   expect(linkElement).toBeInTheDocument();
+// });
```
Comment (cristian-bars) [inline hosni-marco/challenge/toh-fullstack/client/my-toh-redux/src/App.test.js:8]: Mejor no tener codigo comentado

**Review by cristian-bars — APPROVED** [review]
Buen trabajo Hosni!

**Review by SantiagoMC52 — APPROVED** [review]

**Pull request merged**